CoreScriptsRoot/CoreScripts/NotificationScript2.lua

```diff
@@ -0,0 +1,472 @@
+--[[
+		Filename: NotificationScript2.lua
+		Written by: jmargh
+		Description: Handles notification gui for roblox in game events
+--]]
+
+--[[ Services ]]--
+local BadgeService = game:GetService('BadgeService')
+local GuiService = game:GetService('GuiService')
+local Players = game:GetService('Players')
+local PointsService = game:GetService('PointsService')
+local MarketplaceService = game:GetService('MarketplaceService')
+local TeleportService = game:GetService('TeleportService')
+local Settings = UserSettings()
+local GameSettings = Settings.GameSettings
+
+--[[ Script Variables ]]--
+local LocalPlayer = nil
+while not Players.LocalPlayer do
+	wait()
+end
+LocalPlayer = Players.LocalPlayer
+local RbxGui = script.Parent
+local NotificationQueue = {}
+local OverflowQueue = {}
+local FriendRequestBlacklist = {}
+local CurrentGraphicsQualityLevel = nil
+local BindableEvent_SendNotification = Instance.new('BindableFunction')
+BindableEvent_SendNotification.Name = "SendNotification"
+BindableEvent_SendNotification.Parent = RbxGui
+
+--[[ Constants ]]--
+local BG_TRANSPARENCY = 0.7
+local MAX_NOTIFICATIONS = 3
+local NOTIFICATION_Y_OFFSET = 64
+local IMAGE_SIZE = 48
+local EASE_DIR = Enum.EasingDirection.InOut
+local EASE_STYLE = Enum.EasingStyle.Sine
+local TWEEN_TIME = 0.35
+local DEFAULT_NOTIFICATION_DURATION = 5
+
+--[[ Images ]]--
+local PLAYER_POINTS_IMG = 'http://www.roblox.com/asset?id=155363793'
+local BADGE_IMG = 'http://www.roblox.com/asset?id=177200377'
+local FRIEND_IMAGE = 'http://www.roblox.com/thumbs/avatar.ashx?userId='
+local GRAPHICS_QUALITY_LEVELS = 10
+local GRAPHICS_QUALITY_TO_INT = {
```

Comment:
Why not just use the Enum's .Value field?
For SavedQualitySetting it'll return the same value as predicted by this table.